packages/compiler/core/checker.ts

```diff
@@ -2135,7 +2136,7 @@ export function createChecker(program: Program): Checker {
       case "Model":
         clone = finishType({
           ...type,
-          properties: additionalProps.hasOwnProperty("properties")
+          properties: Object.prototype.hasOwnProperty.call(additionalProps, "properties")
```

Comment:
I think I remember @bterlson not thinking this rule was worthwhile for us.

Reply:
Don't have a strong opinion on this one if we want to disable it, just seems like it could prevent some unexpected bug if we do some protype manipulation(which I think we are doing somewhere too) or use the wrong names of properties